course/models.py

```diff
@@ -539,7 +539,10 @@ def tag_user(self, user, tag):
         UserTagging.objects.create(tag=tag, user=user.userprofile, course_instance=self)
 
     def get_enrollment_for(self, user):
-        return Enrollment.objects.filter(course_instance=self, user_profile=user.userprofile).first()
+        try:
+            return Enrollment.objects.get(course_instance=self, user_profile=user.userprofile)
+        except ObjectDoesNotExist:
```

Comment:
Did you consider `Enrollment.DoesNotExist`?

Reply:
Yes, it is absolutely the best practice to use the DoesNotExist attribute from the target model class. That should be clearly highlighted in the official Django docs on making basic database queries.
https://docs.djangoproject.com/en/2.2/topics/db/queries/#retrieving-a-single-object-with-get

Reply:
Thanks, will fix.